opentelemetry-exporter-geneva/geneva-uploader/src/config_service/client.rs

```diff
@@ -0,0 +1,536 @@
+            } else {
+                Err(GenevaConfigClientError::InternalError(
+                    "Cache was not properly updated".to_string(),
+                ))
+            }
+        }
+    }
+
+    /// Internal method that actually fetches data from Geneva Config Service
+    async fn fetch_ingestion_info(&self) -> Result<(IngestionGatewayInfo, MonikerInfo)> {
+        let agent_identity = "GenevaUploader"; // TODO make this configurable
+        let agent_version = "0.1"; // TODO make this configurable
+        let identity = format!(
+            "Tenant=Default/Role=GcsClient/RoleInstance={}",
+            agent_identity
+        );
+
+        let encoded_identity = general_purpose::STANDARD.encode(&identity);
+        let version_str = format!("Ver{}v0", self.config.config_major_version);
+
+        // Construct the following URL:
+        // https://<endpoint>/api/agent/v3/<environment>/<account>/MonitoringStorageKeys/
+        //   ?Namespace=<namespace>
+        //   &Region=<region>
+        //   &Identity=<base64-encoded identity>
+        //   &OSType=<os_type>
+        //   &ConfigMajorVersion=Ver<major_version>v0
+        //   &TagId=<uuid>
+        let endpoint = self.config.endpoint.trim_end_matches('/');
+        let tag_id = Uuid::new_v4().to_string(); //TODO - uuid is costly, check if counter is enough?
+        let mut url = String::with_capacity(endpoint.len() + 200); // Pre-allocate with reasonable capacity
+        url.push_str(endpoint);
+        url.push_str("/api/agent/v3/");
+        url.push_str(&self.config.environment);
+        url.push('/');
+        url.push_str(&self.config.account);
+        url.push_str("/MonitoringStorageKeys/?Namespace=");
+        url.push_str(&self.config.namespace);
+        url.push_str("&Region=");
+        url.push_str(&self.config.region);
+        url.push_str("&Identity=");
+        url.push_str(&encoded_identity);
+        url.push_str("&OSType=");
+        url.push_str(get_os_type());
+        url.push_str("&ConfigMajorVersion=");
+        url.push_str(&version_str);
+        url.push_str("&TagId=");
+        url.push_str(&tag_id);
+
+        let req_id = Uuid::new_v4().to_string();
+
+        let response = self
+            .http_client
+            .get(&url)
+            .header(
+                "User-Agent",
+                format!("{}-{}", agent_identity, agent_version),
+            )
+            .header("x-ms-client-request-id", req_id)
+            .header("Accept", "application/json")
+            .send()
+            .await
+            .map_err(GenevaConfigClientError::Http)?;
+
+        let status = response.status();
+        let body = response.text().await?;
+
+        if status.is_success() {
+            let parsed = match serde_json::from_str::<GenevaResponse>(&body) {
+                Ok(response) => response,
+                Err(_) => {
+                    return Err(GenevaConfigClientError::AuthInfoNotFound(
```

Comment:
Why are we using `GenevaConfigClientError::AuthInfoNotFound` here? This would be some deserialization error, right?

Reply:
Yes, it is deserialization error, if the response can't be converted into `GenevaResponse` object, which actually means that auth info couldn't be obtained from the server.

Reply:
If we are following this convention, then let's be consistent. Let's remove the `SerdeJson` variant altogether. We have another case of deserializing the JWT payload where we return a `SerdeJson` error. Let's return a `JwtTokenError` with the message "Failed to parse {error}".